posthog/session_recordings/queries/session_replay_events.py

```diff
@@ -165,6 +165,127 @@ def get_events(
+            {
+                "team_id": team.pk,
+                "session_id": session_id,
+            },
+        )
+        return [{"event": e[0], "timestamp": e[1]} for e in events]
+
+    def get_similar_recordings(
+        self, session_id: str, team: Team, limit: int = 10, similarity_range: float = 0.5
+    ) -> list[dict]:
+        """Find recordings with similar event sequences.
+        The similarity is based on:
+        1. Having similar event patterns (types and order)
+        2. Similar activity metrics (clicks, keypresses, etc)
+        3. Similar duration
+
+        Args:
+            session_id: The session ID to find similar recordings for
+            team: The team the recording belongs to
+            limit: Maximum number of similar recordings to return
+            similarity_range: How similar the recordings should be (0.0 to 1.0). Default 0.5 means within 50% range
+        """
+        # First get the target recording's metadata
+        target_metadata = self.get_metadata(session_id=session_id, team=team)
+        if not target_metadata:
+            return []
+
+        # Get target recording's events
+        target_events = self.get_events_for_session(session_id, team)
+        if not target_events:
+            return []
+
+        # Build event pattern string for comparison
+        target_pattern = ",".join([e["event"] for e in target_events])
+
+        query = """
+        SELECT
+            sre.session_id,
+            sre.distinct_id,
+            min_first_timestamp,
+            max_last_timestamp,
+            click_count,
+            keypress_count,
+            mouse_activity_count,
+            active_milliseconds,
+            arrayStringConcat(groupArray(e.event), ',') as event_pattern
+        FROM session_replay_events sre
+        JOIN events e ON e.team_id = sre.team_id AND e.$session_id = sre.session_id
+        WHERE sre.team_id = %(team_id)s
+        AND sre.session_id != %(session_id)s
+        GROUP BY
+            sre.session_id,
+            sre.distinct_id,
+            sre.min_first_timestamp,
+            sre.max_last_timestamp,
+            sre.click_count,
+            sre.keypress_count,
+            sre.mouse_activity_count,
+            sre.active_milliseconds
+        HAVING
+            -- Similar activity levels (within similarity_range)
+            abs(click_count - %(target_clicks)s) <= greatest(%(target_clicks)s * %(similarity_range)s, 5) AND
+            abs(keypress_count - %(target_keypresses)s) <= greatest(%(target_keypresses)s * %(similarity_range)s, 5) AND
+            abs(mouse_activity_count - %(target_mouse)s) <= greatest(%(target_mouse)s * %(similarity_range)s, 5) AND
+            abs(active_milliseconds - %(target_active_ms)s) <= greatest(%(target_active_ms)s * %(similarity_range)s, 5000) AND
+            -- Similar event pattern using Levenshtein distance
+            length(event_pattern) > 0 AND
+            event_pattern != %(target_pattern)s AND
```

Comment:
you can _maybe_ generate the target pattern as part of the query
but whether that would be better or worse performance is an entirely different question 🤣 

i also wonder if we could cache this to avoid running it too much... there must be something in recording metadata we can use to key the cache on... like last timestamp or something